### ISSUE TITLE
Make admin override option harder to use

### ISSUE THREAD

**Issue by segevd-opti** [Reviewable/Reviewable#1141]
Currently, the only indication a merge is utilizing the admin override option (bypass branch protections in GH lingo) is a small red triangle (:small_red_triangle:) on the button.
![image](https://github.com/Reviewable/Reviewable/assets/130683943/b5e1fb91-82ec-4f1c-a9f8-108ad5484a98)

In my opinion this protection measure is not clear enough and prone to mistakes.
When merging a PR from GH, the user must check a checkbox in order to make that option available.
![image](https://github.com/Reviewable/Reviewable/assets/130683943/be5f09da-dc3a-4b12-abe7-4a5815a7f302)
This makes the user more aware of what they're doing and less prone to merge by mistake.

Can reviewable introduce such a measure as well?

**Comment by pkaminski** [comment]
Yes!  We have a great design that takes care of this (and more), it's just blocking on some foundational pieces that need to be built first.
